File: app/src/main/java/com/xenous/athenekotlin/storage/Storage.java
package com.xenous.athenekotlin.storage;

import com.xenous.athenekotlin.data.Category;
import com.xenous.athenekotlin.data.Word;

import java.util.ArrayList;

public class Storage {

    private static Storage instance;

    private static ArrayList<Word> wordArrayList = new ArrayList<>();
    private static ArrayList<Word> checkingWordsArrayList = new ArrayList<>();

    private static ArrayList<Category> categoryArrayList = new ArrayList<>();

    public static Storage getInstance() {
        if(instance == null) {
            instance = new Storage();
        }

        return instance;
    }

    public ArrayList<Word> getWordArrayList() {
        return wordArrayList;
    }

    public void setWordArrayList(ArrayList<Word> wordArrayList) {
        Storage.wordArrayList = wordArrayList;
    }

    public ArrayList<Word> getCheckingWordsArrayList() {
        return checkingWordsArrayList;
    }

    public void setCheckingWordsArrayList(ArrayList<Word> checkingWordsArrayList) {
        Storage.checkingWordsArrayList = checkingWordsArrayList;
    }

    public ArrayList<Category> getCategoryArrayList() {
        return categoryArrayList;
    }

    public void setCategoryArrayList(ArrayList<Category> categoryArrayList) {
        Storage.categoryArrayList = categoryArrayList;
    }
}

File: app/src/main/java/com/xenous/athenekotlin/storage/Storage.kt
package com.xenous.athenekotlin.storage

import com.xenous.athenekotlin.data.Category
import com.xenous.athenekotlin.data.Word

var storedInStorageWordsArrayList = mutableListOf<Word>()
var checkingWordsArrayList = mutableListOf<Word>()

var storedInStorageCategoriesArrayList = mutableListOf<Category>()

fun getCategoriesArrayListWithDefault(): MutableList<Category> {
    val defaultList = mutableListOf<Category>()
    defaultList.add(Category("без категории"))
    defaultList.add(Category("спорт"))
    defaultList.add(Category("наука"))
    defaultList.add(Category("еда"))
    defaultList.add(Category("транспорт"))
    defaultList.add(Category("одежда"))
    defaultList.addAll(storedInStorageCategoriesArrayList)

    return defaultList
}